sdk/tests/iot/adu/test_az_iot_adu.c

```diff
@@ -814,12 +817,8 @@ static void test_az_iot_adu_client_parse_service_properties_succeed(void** state
   assert_int_equal(az_span_size(request.workflow.retry_timestamp), 0);
 
   // Update Manifest
-  int32_t out_size;
-  assert_int_equal(
-      az_json_string_unescape(
-          request.update_manifest, (char*)scratch_buffer, sizeof(scratch_buffer), &out_size),
-      AZ_OK);
-  request.update_manifest = az_span_create(scratch_buffer, out_size);
+  request.update_manifest
+      = az_json_string_unescape(request.update_manifest, request.update_manifest);
```

Comment:
Why are we sending the same parameter twice? It looks like we do that in every place we're using this except the tests

Reply:
The signature for the unescape is 
```c
<az_span with resulting buffer and size> az_json_string_unescape(
                   <input to be unescaped>,
                   <output buffer where to put the unescaped string>);
```
So in this case, because we're doing it in place, the input is the same as the output.

Reply:
oh okay, so the function works for in place or not in place unescaping - that makes sense!

Reply:
Yup makes it flexible!